### PR TITLE
Update docs to mention new logging package and the standard Go testing framework

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: DEV-CONVENTIONS.md [hunk 1]
```diff
@@ -46,43 +46,60 @@ Others docs you should look at:
...
-    - More info about system state changes
-
-  - glog.V(4) - Debug level verbosity (for now)
-    - Logging in particularly thorny parts of code where you may want to come
-    back later and check it
-
-As per the comments, the practical production level at runtime is V(2). Developers and QE
-environments may wish to run at V(3) or V(4). If you wish to change the log
-level, you can pass in `-v=X` where X is the desired maximum level to log.
+Istio code is expected to use the standard [Istio logging package](https://godoc.org/istio.io/istio/pkg/log).
```
Comment (kyessenov) [inline DEV-CONVENTIONS.md]: How do we map underlying library logging to zap? Are we going to provide logging adapters?
How do I translate glog levels into istio logging?

File: DEV-CONVENTIONS.md [hunk 2]
```diff
@@ -46,43 +46,60 @@ Others docs you should look at:
...
+When importing external package to Istio, if these packages send logs to the standard Go "log" package or
+to the global [zap logger](https://godoc.org/go.uber.org/zap), the log data will automatically be captured
+by the Istio log package and merged into the primary log stream(s) produced for the component. If
+other logging packages are used, their output will uncoordinated and unrelated to the primary log
```
Comment (kyessenov) [inline DEV-CONVENTIONS.md]: This seems like a major disadvantage of this change. I don't get to make choices for logging used in my libraries.

File: DEV-CONVENTIONS.md [hunk 3]
```diff
@@ -46,43 +46,60 @@ Others docs you should look at:
...
-  - glog.V(4) - Debug level verbosity (for now)
-    - Logging in particularly thorny parts of code where you may want to come
-    back later and check it
-
-As per the comments, the practical production level at runtime is V(2). Developers and QE
-environments may wish to run at V(3) or V(4). If you wish to change the log
-level, you can pass in `-v=X` where X is the desired maximum level to log.
+Istio code is expected to use the standard [Istio logging package](https://godoc.org/istio.io/istio/pkg/log).
+When importing external package to Istio, if these packages send logs to the standard Go "log" package or
+to the global [zap logger](https://godoc.org/go.uber.org/zap), the log data will automatically be captured
```
Comment (kyessenov) [inline DEV-CONVENTIONS.md]: I don't know any serious project that serious standard go `log`. `zap` is a new kid on the block. Most use either `logrus` or `glog`.

File: DEV-CONVENTIONS.md (hunk 2 above)
Comment (geeknoid) [reply DEV-CONVENTIONS.md]: ?

Libraries can use any logging package they want. There's no difference with this new approach. However, if you use go "log" or zap, the logs your package emits will be captured by the Istio logger and rolled into the main log files and participate in log rotation, etc.

If you use a random library for logging, no behavior will change. If you use one of the ones we recommend, the user will get a better experience.

If there are other popular libraries that we want to capture the output, we can augment the Istio package to do so.

glog unforfunately doesn't support having its output captured.

File: DEV-CONVENTIONS.md (hunk 1 above)
Comment (geeknoid) [reply DEV-CONVENTIONS.md]: Are you asking for docs? I don't really understand your question.

Comment (kyessenov) [reply DEV-CONVENTIONS.md]: Imagine, I'm tracing pilot code path. With glog, I can see each step as it flows through envoy, to pilot, to kubernetes, to etcd, and back. With a new library, I only get to see the first bits, and then it's glog. I'd hope at the very least, that we can redirect glog output to istio logger. Does zap provide such a feature?

**Comment by geeknoid** [comment]
glog doesn't allow its output to be captured in any way unfortunately.
People have submitted PRs to enable this, but glog is specifically not
taking PRs.

**Comment by kyessenov** [comment]
Can we fork glog, and rewire godep import to an internal zap-compatible
package?

**Comment by geeknoid** [comment]
I don't know about the godep voodoo, but conceptually this would be pretty
simple.

**Pull request merged**